packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  // The envelope to be sent
+  envelope: Envelope;
+  // A function that returns an event from the envelope if one exists
+  getEvent(): Event | undefined;
+}
+
+type Matcher = (param: MatchParam) => string[];
+
+function eventFromEnvelope(env: Envelope): Event | undefined {
+  let event: Event | undefined;
+
+  forEachEnvelopeItem(env, (item, type) => {
+    if (type === 'event') {
+      event = Array.isArray(item) ? (item as EventItem)[1] : undefined;
+    }
+    // bail out if we found an event
+    return !!event;
+  });
+
+  return event;
+}
+
+/**
+ * Creates a transport that can send events to different DSNs depending on the envelope contents.
+ */
+export function makeMultiplexedTransport<TO extends BaseTransportOptions>(
+  createTransport: (options: TO) => Transport,
+  matcher: Matcher,
+): (options: TO) => Transport {
+  return options => {
+    const fallbackTransport = createTransport(options);
+    const otherTransports: Record<string, Transport> = {};
+
+    function getTransport(dsn: string): Transport {
+      if (!otherTransports[dsn]) {
+        const url = getEnvelopeEndpointWithUrlEncodedAuth(dsnFromString(dsn));
```

Comment:
do we need error handling on the DSN here?

Reply:
`dsnFromString` already throws for invalid DSN. We could catch and re-throw but I'm not sure we'd gain much as the stack trace should be reasonable. For example, here is what I saw while creating the tests:

```
    SentryError: Invalid Sentry Dsn: h

      37 |
      38 |   if (!match) {
    > 39 |     throw new SentryError(`Invalid Sentry Dsn: ${str}`);
         | ^
      40 |   }
      41 |
      42 |   const [protocol, publicKey, pass = '', host, port = '', lastPath] = match.slice(1);

      at Object.utils_1 [as dsnFromString] (../utils/src/dsn.ts:39:1)
      at getTransport (src/transports/multiplexed.ts:51:59)
      at map (src/transports/multiplexed.ts:63:73)
          at Array.map (<anonymous>)
      at Object.<anonymous> (src/transports/multiplexed.ts:63:58)
```